boris_tests/BBQ_test.go

```diff
@@ -0,0 +1,110 @@
+package boris_tests
+
+import (
+	"errors"
+	"testing"
+
+	mock_beeffarm "github.com/artiomgiza/test_me/pkgs/beef-farm/mock"
+	mock_chickenfarm "github.com/artiomgiza/test_me/pkgs/chicken-farm/mock"
+	mock_coalfarm "github.com/artiomgiza/test_me/pkgs/coal-farm/mock"
+	mock_mangalstore "github.com/artiomgiza/test_me/pkgs/mangal-store/mock"
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
+)
+
+var ctrl *gomock.Controller
+var mockBeef *mock_beeffarm.MockProvider
+var mockChicken *mock_chickenfarm.MockProvider
+var mockCoal *mock_coalfarm.MockProvider
+var mockMangal *mock_mangalstore.MockProvider
+
+func setup(t *testing.T) func() {
+	ctrl = gomock.NewController(t)
+	mockBeef = mock_beeffarm.NewMockProvider(ctrl)
+	mockChicken = mock_chickenfarm.NewMockProvider(ctrl)
+	mockMangal = mock_mangalstore.NewMockProvider(ctrl)
+	mockCoal = mock_coalfarm.NewMockProvider(ctrl)
+	Instance = coolPriceCalculator{
```

Comment:
`Instance` is interface
The test should test concrete implementation, o might be define "object under test" in the test file?

Reply:
not sure I understand. Interface is a variable of type Provider, which is the interface. If I am missing anything, let's talk tomorrow

Reply:
If you have not only public but also **private** methods - you can't test it with `Instance` variable.
This is the reason there is `var subject coolPriceCalculator` defined in other examples.
This is object under test, concrete implementation